promql/prettier/node.go

```diff
@@ -89,101 +86,16 @@ func (n *nodeInfo) parentNode(head parser.Expr, rnge parser.PositionRange) parse
-	var nodeMatch bool
-	switch node := head.(type) {
-	case *parser.ParenExpr:
-		if node.PositionRange().Contains(posRange) {
-			nodeMatch = true
-			stack = append(stack, node)
-			if _stack, _node, found := n.nodeHistory(node.Expr, posRange, stack); found {
-				head = _node
-				stack = _stack
-			}
-		}
-	case *parser.VectorSelector:
-		if node.PositionRange().Contains(posRange) {
-			nodeMatch = true
-			stack = append(stack, node)
-		}
-	case *parser.BinaryExpr:
-		if node.PositionRange().Contains(posRange) {
-			nodeMatch = true
-			stack = append(stack, node)
-			stmp, _node, found := n.nodeHistory(node.LHS, posRange, stack)
-			if found {
-				return stmp, _node, found
-			}
-			if stmp, _node, found = n.nodeHistory(node.RHS, posRange, stack); found {
-				return stmp, _node, found
-			}
-			// Since the item exists in both the child. This means that it is in binary expr range,
-			// but not satisfied by a single child. This is possible only for Op and grouping
-			// modifiers.
-			if node.PositionRange().Contains(posRange) {
-				return stack, head, true
-			}
-		}
-	case *parser.AggregateExpr:
-		if node.PositionRange().Contains(posRange) {
-			nodeMatch = true
-			stack = append(stack, node)
-			stmp, _head, found := n.nodeHistory(node.Expr, posRange, stack)
-			if found {
-				return stmp, _head, true
-			}
-			if node.Param != nil {
-				if stmp, _head, found = n.nodeHistory(node.Param, posRange, stack); found {
-					return stmp, _head, true
-				}
-			}
-		}
-	case *parser.Call:
-		if node.PositionRange().Contains(posRange) {
-			nodeMatch = true
-			stack = append(stack, node)
-			for _, exprs := range node.Args {
-				if exprs.PositionRange().Contains(posRange) {
-					if stmp, _head, found := n.nodeHistory(exprs, posRange, stack); found {
-						return stmp, _head, true
-					}
-				}
-			}
-		}
-	case *parser.MatrixSelector:
-		if node.VectorSelector.PositionRange().Contains(posRange) {
-			stack = append(stack, node)
-			nodeMatch = true
-			if _stack, _node, found := n.nodeHistory(node.VectorSelector, posRange, stack); found {
-				stack = _stack
-				head = _node
-			}
-		}
-	case *parser.UnaryExpr:
-		stack = append(stack, node)
-		if node.Expr.PositionRange().Contains(posRange) {
-			nodeMatch = true
-			if _stack, _node, found := n.nodeHistory(node.Expr, posRange, stack); found {
-				stack = _stack
-				head = _node
-			}
-		}
-	case *parser.SubqueryExpr:
-		if node.Expr.PositionRange().Contains(posRange) {
-			nodeMatch = true
-			stack = append(stack, node)
-			if _stack, _node, found := n.nodeHistory(node.Expr, posRange, stack); found {
-				stack = _stack
-				head = _node
-			}
-		}
-	case *parser.NumberLiteral, *parser.StringLiteral:
-		if node.PositionRange().Contains(posRange) {
-			nodeMatch = true
-			stack = append(stack, node)
+func (n *nodeInfo) nodeHistory(head parser.Node, posRange parser.PositionRange, stack []parser.Node) ([]parser.Node, parser.Node) {
```

Comment:
Both passing the `stack` and `head` as a reference and returning those references, seems redundant.

Reply:
Oh yes, they are not required. I will remove them too. Thank you.